x-pack/platform/packages/shared/kbn-ai-assistant/src/hooks/use_knowledge_base.tsx

```diff
@@ -29,25 +31,67 @@ export function useKnowledgeBase(): UseKnowledgeBaseResult {
     },
     [service]
   );
+  const [isInstalling, setIsInstalling] = useState(false);
+  const [installError, setInstallError] = useState<Error>();
+
+  useEffect(() => {
+    if (isInstalling && !!statusRequest.value?.endpoint) {
+      setIsInstalling(false);
+    }
+  }, [isInstalling, statusRequest]);
 
   const setupKb = useCallback(async () => {
+    setIsInstalling(true);
+    setInstallError(undefined);
+
+    let attempts = 0;
+    const MAX_ATTEMPTS = 5;
     try {
-      await service.callApi('POST /internal/observability_ai_assistant/kb/setup', {
-        signal: null,
-      });
+      // install
+      await retrySetupIfError();
+
+      if (ml.mlApi?.savedObjects.syncSavedObjects) {
+        await ml.mlApi.savedObjects.syncSavedObjects();
+      }
+
+      // do one refresh to get an initial status
+      await statusRequest.refresh();
     } catch (e) {
+      setInstallError(e);
       notifications!.toasts.addError(e, {
         title: i18n.translate('xpack.aiAssistant.errorSettingUpInferenceEndpoint', {
           defaultMessage: 'Could not create inference endpoint',
         }),
       });
     } finally {
-      statusRequest.refresh();
+      // setIsInstalling(false);
+    }
+
+    async function retrySetupIfError() {
```

Comment:
This should simply be called

```suggestion
    async function setupKnowledgeBase() {
```